presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -291,7 +292,7 @@ public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, N
         public Expression rewriteFunctionCall(FunctionCall node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
         {
             // A grouping() is preceded by a GroupId node when the source node is projecting a groupid symbol
-            if (!source.getOutputSymbols().contains(new Symbol("groupid")) && node.getName().toString().equals(GROUPING)) {
+            if (!source.getOutputSymbols().stream().anyMatch(symbol -> symbol.getName().contains(GROUPID_SYMBOL_HINT)) && node.getName().toString().equals(GROUPING)) {
```

Comment:
@sopel39 checkout 7fc3a816cd9d005d595b62dbb191e38f5d531699. I actually got rid of this piece of code and instead do the literal 0 re-write earlier. I did change the `SymbolToInputRewriter` to not rely on string matching.